### ISSUE TITLE
Enabling all content filters results in extreme startup time on pc with limited power

### ISSUE THREAD

**Issue by Kacper-Kondracki** [brave/brave-browser#34114]
## Description 
Normally, brave startup is fast, but when enabling all available filters, on my machine (Ryzen 3 5300U (4cores/8threads))j startup time increase to around 5 seconds. I would really like to have filters enabled for every language, but the startup delay forces me to make compromise and reduce number of enabled filters

## Steps to Reproduce

   1. Enable all available content filters in settings
   2. Restart Brave

## Actual result:
Slow startup on pc with limited power

## Expected result:
Quick startup, like when using only filters enabled by default

## Reproduces how often: 

Easily on weaker machines, Hard on powerful pc (startup time increase is not noticable)

## Brave version (brave://version info)

Brave	1.60.110 Chromium: 119.0.6045.105 (Official Build) (64-bit) 
Revision	0bb6e7a81aed340b5a9e8b752c1cf92768257c40

## Version/Channel Information:

- Can you reproduce this issue with the current release?
yes 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel?

**Comment by Emi-HoloGhostRevisionist88** [comment]
Brave already has 150K rules by default, you don't need rules for different languages, just in rare cases.
The more you enable, the slower everything and the more problems you will have, that's just about adblocking in general, not exclusive to Brave.

**Comment by Marko-98** [comment]
I don't think that can be fixed. It's completely normal for the browser to slow down after subscribing to a lot of filters. Because it has to load, well, if not millions of rules in a large number of filters. 

I'm not sure why would you want to be subscribed to all filters anyway.

**Comment by Kacper-Kondracki** [comment]
I see this is still open https://github.com/brave/brave-browser/issues/27161
If I'm right, this should really improve startup time overall. 

Even when comparing startup time without any filters and with only these filters + one regional 
![image](https://github.com/brave/brave-browser/assets/43544435/5fcba45a-f526-415a-899c-b194e84595cf)
Startup delay difference can be feelable (less than second longer, although it is fairly acceptable)

Also adblock could be non-blocking, when Brave is just launched with new tab page.
